### DISCUSSION TITLE
daily auto-import from two german bank accounts possible?

### DISCUSSION THREAD

**Discussion by prankousky** [firefly-iii/firefly-iii#7066] [Firefly III]
Hi everybody,

I have been using `firefly-iii` for quite some time now, however, I never figured out a good, reliable way to sync my bank accounts.

The accounts in question are with

* ING-DiBa
* Kreissparkasse Ratzeburg

I had previously used `firefly-iii-fints-importer`, which worked fine. But when I had to re-install firefly, I chose to use the `fireflyiii-data-importer`, which is an officially way to do the same thing (I believe?).

The import function work; I can make use of `FinTS` protocol with both tools, and it works with both my bank accounts (only pulling data *from* my bank accounts has been tested, but that's all I want/need, anyway).

Both these importers had to manually be called, though. So, let's say I'd want to work with (organize, ...) firefly-iii. I'd then

* open a browser tab to firefly-iii
* open a browser tab to firefly-iii-fints-importer **or** fireflyiii-data-importer
* import data from ING-DiBa
* import data from Kreissparkasse Ratzeburg
* reload firefly-iii 

Then there'd still be minor issues:

##### balance issues

sometimes, the account balance would be incorrect; let's say account a **was** `€ 50.00`, and there had been a deposit of `€ 6.00` and a withdrawal of `€ 12.00`; balance should now be `€ 44.00`, but it would be something completely different (*I cannot give an example, as this is currently not reproducible with my setup*).

##### description fields

`description` would usually contain something like this `302.3354415.5653960 AMZN Mktp DE 5YJJKIYGZ37AI4ZZ ` or simply `01.13456.4`.

With one of the tools (not sure which one it was), every `description` field would contain additional text, which I couldn't figure out how to filter out. When this happened, my rules would still work, but it would look weird when manually looking at these balances (because this additional text was a quite long block of characters that just distracted from the actual description).

---

# Main Question

How would you configure `firefly-iii` - and, perhaps, an external tool - to automatically import all transactions from these bank accounts without manually having to do so? 

While doing so, this is important:

* obviously, cross-check transactions, so there are no doubles
* if possible: assure that the account balance displayed in `firefly-iii` is actually **equal to the bank account balance**

---

I am not sure about ING-DiBa, but am rather sure that Kreissparkasse Ratzeburg requires some sort of verification every x months (3 months??). In my case, when I log into my bank account or banking app, it will ask for a freshly generated `ChipTAN` number. 

So if auto-import is possible, there'd have to be some kind of watchdog for this so that my bank login won't get blocked because the importer constantly tries to log in without waiting for this code.

---

What are your suggestions? Thank you in advance for your ideas :)

**Comment by JC5** [comment]
This calls for some advanced scripting work. Either python, PHP, or something else entirely. Do your banks offer some kind of API?

**Comment by piotrcichosz** [comment]
How would you import these 2 banks? If I see correctly, first one is supoprted by Spectre - https://www.saltedge.com/products/account_information/coverage/de?provider_group=ING&provider_code=ing_oauth_client_de
second one by nordigen - https://nordigen.com/en/coverage/

to automate import, I'm doing it right now for myself with nordigen - you configure data-importer and nordigen account, next you do first import through web (data-importer) and at the end you click on "download configuration file"
later on you can use that file to call automatic imports (call them by sending POST request, or triggering docker, or running php artisan import: command.. )
so here you can schedule daily or weekly job to import it automatically - if you configure email you'll receive summary of that import

balance issues - I recommend you check the dates. Payment and processed date can differ, maybe that's why you get different balance?
I started support ticket today with nordigen regarding that, because in bank I see 6 transactions that are "processing", but in nordigen endpoint I have only transactions in "bookied" state - no pending. I'm not sure if this is how my bank implemented PSD2, or nordigen issue, or maybe this is how its supposed to work.
Anyway, play more and when you find problem then investigate further. I imported history since 01.01.2023 and apart of small differences in January (but there I removed some transactions intentionally) - next months are perfect to a cent.

description - for nordigen you can go to Developers - My Account Data, connect to a bank from there and download csv or view transactions.
